perceval/components/core_catalog/heralded_cnot.py

```diff
@@ -61,29 +58,19 @@ def build(self):
             return self.build_processor(backend=self._opt('backend'))
 
     def build_circuit(self, **kwargs):
-        return (Circuit(8, name="Heralded CNOT")
-                .add(1, PERM([2, 4, 3, 0, 1]))
-                .add(4, BS.H())
-                .add(3, PERM([1, 3, 0, 4, 2]))
-                .add(3, BS.H())
-                .add(3, PERM([2, 0, 1]))
-                .add(2, BS.H(theta=self.theta1))
-                .add(4, BS.H(theta=self.theta1))
-                .add(3, PERM([1, 2, 0]))
-                .add(3, BS.H())
-                .add(1, PERM([2, 0, 3, 1, 6, 5, 4]))
-                .add(2, BS.H(theta=self.theta2))
-                .add(2, PERM([1, 0]))
-                .add(4, BS.H(theta=self.theta2))
-                .add(4, PERM([1, 2, 0]))
-                .add(4, BS.H())
-                .add(1, PERM([4, 3, 0, 2, 1])))
+        c = Circuit(6, name="Heralded CNOT")
+        c.add(2, Circuit(2, name='H') // BS.H())
+        heralded_cz = HeraldedCzItem()
+        c.add(0, heralded_cz.build_circuit())
+        c.add(2, Circuit(2, name='H') // BS.H())
+        return c
 
     def build_processor(self, **kwargs):
-        p = self._init_processor(**kwargs)
-        return p.add_port(0, Port(Encoding.DUAL_RAIL, 'ctrl')) \
-            .add_port(2, Port(Encoding.DUAL_RAIL, 'data')) \
-            .add_herald(4, 0) \
-            .add_herald(5, 1) \
-            .add_herald(6, 0) \
-            .add_herald(7, 1)
+        p = Processor(backend="SLOS", m_circuit=4, source=Source())
```

Comment:
You are bypassing the `self._init_processor(**kwargs)` method call.
This means :
* The circuit building code is duplicated
* You hardcoded that you're using SLOS (_init_processor creates a Processor with any backend when kwargs require it)

So, could you get back to how other catalog items create their processors?